### DISCUSSION TITLE
颜色显示存在问题？

### DISCUSSION THREAD

**Discussion by Soniceso** [libnyanpasu/clash-nyanpasu#1154] [Q&A] [answered]
### 复现步骤 / Step to reproduce

更新后感觉深色模式的的颜色有一点奇怪，看了下颜色代码和原来一样都是#1380E4，下载了一个拾色器试了下发现无论是浅色模式还是深色模式是下，软件页面中都不存在设置的主题颜色。
浅色模式
![image](https://github.com/LibNyanpasu/clash-nyanpasu/assets/10915608/141fa2d8-2e8d-4fee-bb1b-d6899a4118bc)
深色模式
![image](https://github.com/LibNyanpasu/clash-nyanpasu/assets/10915608/ffacd37b-b7d2-471e-b05e-1c5d969b16b6)
不太清楚是故意这么做的还是存在问题，毕竟设置颜色的逻辑和原来也稍有不一样了，现在只剩下了主题设置这个选项。

### 预期行为 / Expected behaviour

原来的1.5.1的话，至少猫猫和一些按钮的底色是一样的，而且是设置的主要颜色的代码
![image](https://github.com/LibNyanpasu/clash-nyanpasu/assets/10915608/92df6ab2-cfba-40fb-9dee-91bca9c47544)

### 实际行为 / Actual Behaviour

如上

### Clash Nyanpasu 版本号 / Clash Nyanpasu

1.5.2+d8b605f

### Clash 核心及其版本号 / Clash core and version

1.18.6

### 是否为 Pre-release / Is pre-release version

d8b605f

### 操作系统及版本 / OS version

win11

### 自查步骤 / Verify steps

- [ ] 如果您有足够的时间和能力，并愿意为此提交 PR ，请勾上此复选框 / Pull request is welcome. Check this if you want to start a pull request
- [X] 您已知悉如果没有提供正确的系统信息，以及日记，您的 Issue 会直接被关闭 / You have known that if you don't provide correct system information and logs, your issue will be closed directly
- [X] 您已仔细查看并知情 [Q&A](https://nyanpasu.elaina.moe/zh-CN/others/issues.html) 和 [FAQ](https://nyanpasu.elaina.moe/zh-CN/others/faq.html) 中的内容 / You have read and understood the contents of [Q&A](https://nyanpasu.elaina.moe/others/issues.html) and [FAQ](https://nyanpasu.elaina.moe/others/faq.html).
- [X] 您已搜索过 [Issue Tracker](https://github.com/keiko233/clash-nyanpasu/issues)，没有找到类似内容 / I have searched on [Issue Tracker](https://github.com/keiko233/clash-nyanpasu/issues), No duplicate or related open issue has been found
- [X] 您确保这个 Issue 只提及一个问题。如果您有多个问题报告，烦请发起多个 Issue / Ensure there is only one bug report in this issue. Please make mutiply issue for mutiply bugs
- [X] 您确保已使用最新 Pre-release 版本测试，并且该问题在最新 Pre-release 版本中并未解决 / This bug have not solved in latest Pre-release version

**Comment by keiko233** [comment, marked as answer]
取色器使用material-color-utilities，会对颜色进行hls转换计算从而得到最终颜色，所以只需要输入一个颜色就可以计算出其他颜色

https://github.com/LibNyanpasu/clash-nyanpasu/blob/main/frontend/ui/materialYou/createTheme.ts#L36
